src/frontend/admin/src/components/presentational/files/thumbnail/FileThumbnail.tsx

```diff
@@ -1,15 +1,24 @@
 import * as React from "react";
+import { ReactNode } from "react";
 import Box from "@mui/material/Box";
 import IconButton from "@mui/material/IconButton";
 import Typography from "@mui/material/Typography";
-import { Close, FileOpen } from "@mui/icons-material";
+import ChangeCircleOutlinedIcon from "@mui/icons-material/ChangeCircleOutlined";
+import { ThumbnailDetailField } from "@/services/api/models/Image";
 
 interface Props {
-  file: File;
-  onDelete?: (file: File) => void;
+  file: File | ThumbnailDetailField;
+  onReplace?: () => void;
```

Comment:
Nitpick: in order to be generic with other React's form components WDYT about using `onChange`?